bot/src/main/java/com/community/tools/service/openai/OpenAiService.java

```diff
@@ -0,0 +1,121 @@
+package com.community.tools.service.openai;
+
+import com.community.tools.dto.OpenAiRequestDto;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.UncheckedIOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Collections;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+public class OpenAiService {
+  @Value("${openai.url}")
+  private String url;
+  @Value("${openai.token}")
+  private String apiKey;
+  @Value("${openai.model}")
+  private String model;
+  private final ObjectMapper objectMapper;
+
+  public OpenAiService(ObjectMapper objectMapper) {
+    this.objectMapper = objectMapper;
+  }
+
+  /**
+   * This is the main method in service which processes prompt.
+   * @param prompt users request to the openai api
+   * @return ready-to-use string response from AI
+   */
+  public String processPrompt(String prompt) {
+    try {
+      HttpURLConnection connection = createConnection();
```

Comment:
Can we use [RestTemplate](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/client/RestTemplate.html) to implement the API requests logic? It'll take fewer lines of code and it's much easier to use and understand